### ISSUE TITLE
Ray client suppresses error messages

### ISSUE THREAD

**Issue by robertnishihara** [ray-project/ray#18512]
## Reproduce
1. `ray start --head`
2. Run the following in a local Python interpreter
    ```python
    import ray
    ray.init(address='ray://127.0.0.1:10001')  # To see the correct behavior, replace this with ray.init()

    @ray.remote
    def f():
        raise Exception('error!')

    f.remote()
    ```
3. You won't see any error message, even after waiting!

Related to https://github.com/ray-project/ray/issues/14485 (though that was not specific to Ray client). Not sure if this is a regression or was just never implemented.

cc @AmeerHajAli

**Comment by mwtian** [comment]
Since the exception is the result of `f()`, I think only raising the exceptions with `ray.get(f.remote())` seems like acceptable semantics. For logging the errors without calling `ray.get()`, this is possible in Ray cluster because the exception is stored in the local object store where the script runs. But in Ray client currently, the exception is never returned to the client without calling `ray.get()` first.

**Comment by AmeerHajAli** [comment]
@robertnishihara @ericl , are you OK with the explanation provided by Mingwei above?

**Comment by ericl** [comment]
Is this a bug in client reference counting? From what I understand the object ref returned by f.remote should get gc'ed on the client, hence should be deleted on the server as well. That will trigger the unhandled exception error. If the error isn't printed, it's probably because the reference is still alive somehow.

**Comment by mwtian** [comment]
Ok, I see that we are expecting the error log to be streamed from Ray client-server to client.

Here is what I observe when running Ray client in interactive mode:
- Unhandled exception is printed with
```
a = f.remote()
del a
```
- But only printed after running `2+2` or shutting down interactive shell with
```
f.remote()
```

It seems logic similar to #15432 also needs to be implemented in Ray client.

**Comment by ericl** [comment]
If you add a sleep, does it show up? I don't think anything else should be needed since it is implemented at the server side.

**Comment by ckw017** [comment]
I think the problem here is that the server side doesn't know when the client is being run interactively

**Comment by mwtian** [comment]
Yeah we may need to propagate the `interactive` flag from Ray client initialization to server. Otherwise the logic to log unhandled errors does not seem to get trigger as time passes.

**Comment by cszhu** [comment]
Hello! This P2 issue has seen no activity in at least 2 years. It will be closed in 2 weeks as part of ongoing cleanup efforts.

Please remove the `pending-cleanup` label if you believe this issue should remain open.

Thanks for contributing to Ray!
